discovery.go

```diff
@@ -0,0 +1,17 @@
+package jwtverifier
+
+type Discovery interface {
+	GetWellKnownUrl() string
+}
+
+type OIDC struct {
+	wellKnownUrl string
+}
+
+func NewOIDCDiscovery() Discovery {
+	return &OIDC{wellKnownUrl: "/.well-known/openid-configuration"}
+}
+
+func (d *OIDC) GetWellKnownUrl() string {
+	return d.wellKnownUrl
+}
```

Comment:
Same as adaptor.  the interfaces were split out so people could create their own discovery type as well.

Reply:
Shouldn't this package just work with datadog's method of verification? I don't really understand why people should be implementing their own stuff to use datadog auth.

Reply:
So, golang currently only supports `OIDC` but there is also a discovery method for `OAuth`.... The discovery method for the wellKnownURL for OIDC is:

`wellKnownUri = "/.well-known/openid-configuration"`

While the wellKnownURL for OAuth is:

`wellKnownUri = "/.well-known/oauth-authorization-server"`